cns/client/client_test.go

```diff
@@ -2145,3 +2159,98 @@ func TestNMASupportedAPIs(t *testing.T) {
 		})
 	}
 }
+
+func TestGetAllNCsFromCns(t *testing.T) {
+	emptyRoutes, _ := buildRoutes(defaultBaseURL, clientPaths)
+	exp := cns.GetAllNetworkContainersResponse{
+		NetworkContainers: nil,
+		Response: cns.Response{
+			ReturnCode: types.Success,
+			Message:    "",
+		},
+	}
+	mockDo := &mockdo{
+		errToReturn:            nil,
+		objToReturn:            cns.GetAllNetworkContainersResponse{},
+		httpStatusCodeToReturn: http.StatusOK,
+	}
+	client := &Client{
+		client: mockDo,
+		routes: emptyRoutes,
+	}
+
+	got, err := client.GetAllNCsFromCns(context.TODO())
+	assert.NoError(t, err)
+	assert.Equal(t, exp.Response.ReturnCode, got.Response.ReturnCode)
+}
+
+func TestPostAllNCsToCns(t *testing.T) {
+	// Create the routes necessary for a test client
+	emptyRoutes, _ := buildRoutes(defaultBaseURL, clientPaths)
+
+	createNCTests := []struct {
+		name      string
+		client    *RequestCapture
+		req       cns.PostNetworkContainersRequest
+		expReq    *cns.PostNetworkContainersRequest
+		shouldErr bool
+	}{
+		{
+			"Empty request",
+			&RequestCapture{
+				Next: &mockdo{},
+			},
+			cns.PostNetworkContainersRequest{},
+			nil,
+			true,
+		},
+		{
+			"Happy path",
+			&RequestCapture{
+				Next: &mockdo{},
+			},
+			postAllNcsRequests,
+			&postAllNcsRequests,
+			false,
+		},
+	}
+
+	for _, test := range createNCTests {
+		test := test
+		t.Run(test.name, func(t *testing.T) {
+			t.Parallel()
+			client := &Client{
+				client: test.client,
+				routes: emptyRoutes,
+			}
+
+			err := client.PostAllNCsToCns(context.TODO(), test.req)
+			if err != nil && !test.shouldErr {
+				t.Fatal("unexpected error: err:", err)
+			}
+
+			if err == nil && test.shouldErr {
+				t.Fatal("expected an error but received none")
+			}
+
+			// Make sure if we expected a request, the correct one will be received
+			if test.expReq != nil {
+				// Make sure a request was actually received
+				if test.client.Request == nil {
+					t.Fatal("expected to receive a request, but none received")
+				}
+
+				// Decode the received request for later comparison
+				var gotReq cns.PostNetworkContainersRequest
+				err = json.NewDecoder(test.client.Request.Body).Decode(&gotReq)
+				if err != nil {
+					t.Fatal("error decoding received request: err:", err)
+				}
+
+				for i := 0; i < len(test.expReq.CreateNetworkContainerRequests); i++ {
+					assert.Equal(t, test.expReq.CreateNetworkContainerRequests[i], gotReq.CreateNetworkContainerRequests[i])
```

Comment:
Is this stable against different orderings of the elements of the slice? You may have to sort here first...

Reply:
The order will affect the results. So it needs to be sorted at first.